services/catarse/app/models/payment.rb

```diff
@@ -31,6 +32,12 @@ class Payment < ApplicationRecord
       .having('count(gp.id) < payments.installments')
   }
 
+  def updated_slip_fee
+    if self.payment_method == "BoletoBancario" && !self.project.is_sub? && self.project.service_slip_fee > 0 && self.slip_fee == 0
+      self.update(slip_fee: self.project.service_slip_fee)
+    end
+  end
+
```

Comment:
**Sugestão:**
- Acho que ficaria melhor essa taxa ser preenchida na criação do pagamento lá no Catarse Pagarme, assim tiraria esse callback do modelo.
- Essa linhá tá muito longa também. Sugiro colocar nas configs do teu vscode uma régua que meça as 120 colunas:
```json
"editor.rulers": [120],
"workbench.colorCustomizations": {
  "editorRuler.foreground": "#ffffff30"
},
```
- Os `self` não são necessários nesse código.
- O `self.payment_method == "BoletoBancario"` pode ser trocado pelo `slip_payment?` pois há um método desse no payment.
- Se usar `before_save` pra isso, terá problemas, pois um boleto que for gerado antes do deploy, ele não terá a taxa de R$ 2,00, e ele for pago após o deploy, esse callback fará com que ele fique com a taxa de R$ 2,00 contabilizada, mas o apoiador não terá pago ela.

Reply:
Pelo que entendi aqui, como teremos a regra do `after create project` esse problema que você colocou aqui deixa de existir correto @stephannv ? Pois só é possível gerar um boleto para um projeto publicado, e como vamos usar a regra do `after create project`, não será possível ocorrer esse cenário que você descreveu, de um boleto ser gerado antes do deploy.

Reply:
Isso @thiagocatarse , eu acabei comentando mesmo assim por não ter certeza de como iria ficar, aí já ficava registrado aqui.

Reply:
stephann sobre o uso do self, eu realizei as mudanças conforme o robocup e ficou assim:
```
if slip_payment? && !project.is_sub? && project.service_slip_fee.positive? && self.slip_fee.zero?
   self.slip_fee = project.service_slip_fee
end
```
Quando eu tentei remover o self. do slip_fee ocorria um erro onde a aplicação travava ou demorava muito até aparecer uma mensagem de erro na página. Não entendi ao certo, mas investigando com byebug descobri que esse erro só acontecia com a remoção do self. e o próprio rubocop chamava a atenção quando removia o self. nesse caso.